rust/yarp-sys/src/lib.rs

```diff
@@ -0,0 +1,36 @@
+#![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+
+use std::ffi::c_char;
+
+// In `build.rs`, we use `bindgen` to generate bindings based on C headers and `librubyparser`. Here
+// is where we pull in those bindings and make them part of our library.
+include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+
+pub type yp_parser_t_lex_modes = yp_parser__bindgen_ty_1;
+
+//  ╭──────────────────────────────────────────────────────────────────────────────╮
+//  │ Hand-rolling `yp_string_t` things                                            │
+//  │                                                                              │
+//  │ `bindgen` can generate these, but because of the nested `union`s, the        │
+//  │ types end up with weird names. I'm sure there's a way to fix that, but I     │
+//  │ haven't been able to. So I'm hand-rolling these for now.                     │
+//  ╰──────────────────────────────────────────────────────────────────────────────╯
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub struct yp_string_t {
+    pub type_: yp_string_t_type,
+    pub source: *const c_char,
+    pub length: usize,
+}
+
+#[repr(u32)]
+#[non_exhaustive]
+#[derive(Clone, Copy)]
+pub enum yp_string_t_type {
+    YP_STRING_SHARED,
+    YP_STRING_OWNED,
+    YP_STRING_CONSTANT,
+    YP_STRING_MAPPED,
+}
```

Comment:
can we also derive `Debug` for these types?

Reply:
Yup.